### PR TITLE
Features/filtering hashtags 2

### PR DESCRIPTION
**Pull request by shannonwells**
## Purpose
[Finishes #179255870](https://www.pivotaltracker.com/story/show/179255870)

## Solution
Make Foodee display only items tagged with what's in the linked story.  This is **NOT** a feature to be merged back to example-client. It is **ONLY** for the demo.

## Change summary
* Filtering done via .env environment variables.
* Populate script adds some tags to some posts.
* Filtering is done within PostList
* We strip any leading "#" characters.

Steps to Verify
----------------
1. Copy .env.sample to .env to get the final list of tags to filter on.
2.  Set up Foodee to run against a local hardhat note, and run the static-server, and run the updated populate.mjs script to add events with demo-relevant content. 
3. Login to Foodee.
4. There shouldn't be any posts showing in All Posts that don't have tags that are in the list.

Additional details / screenshot
----------------
- Any supplemental pictures or material
- 
Screen Shot 2021-08-17 at 11 26 07 AM

### CODE REVIEW

File: scripts/populate.mjs [hunk 1]
```diff
@@ -53,7 +59,12 @@ const accounts = [
         "https://brooklynfarmgirl.com/wp-content/uploads/2015/06/Mexican-Taco-Salsa-Hot-Dog_12.jpg"
       ),
     ],
-    text: "Hot take: Hotdogs aren't a sandwich, they're a taco",
+    text: "Hot take: A hotdog's not a sandwich, it's a taco",
```
Comment (shannonwells) [inline scripts/populate.mjs:62]: This one should NOT show up in the feed.

File: scripts/populate.mjs [hunk 2]
```diff
@@ -42,6 +44,10 @@ const accounts = [
         "https://www.dietandi.com/wp-content/uploads/2011/11/Eating-Cereal.jpg"
       ),
     ],
+    tag: [
+      {name: "cereal"},
```
Comment (shannonwells) [inline scripts/populate.mjs:48]: should not show up in the feed.

File: scripts/populate.mjs [hunk 3]
```diff
@@ -193,7 +208,13 @@ const accounts = [
         "https://image.shutterstock.com/image-photo/british-blue-kitten-very-beautiful-260nw-796071583.jpg"
       ),
     ],
-    text: "My favorite cartoon is Spongebob",
+    text: "I really love sushi!!",
+    tag: [
+      {name: "sushi"},
```
Comment (shannonwells) [inline scripts/populate.mjs:213]: should show in the feed.

File: scripts/populate.mjs [hunk 4]
```diff
@@ -207,6 +228,7 @@ const accounts = [
     ],
     text:
       "The pizza shop down the street is giving out free donuts. Kinda sketchy",
+    tag: [{name: "pizza"},{name: "cucina"},{name: "Italian"},{name: "sketch"},{name: "restaurant"}]
```
Comment (shannonwells) [inline scripts/populate.mjs:231]: should show in the feed.

File: scripts/populate.mjs [hunk 5]
```diff
@@ -266,6 +288,7 @@ const accounts = [
       ),
     ],
     text: "My favorite cartoon is Spongebob",
+    tag: {name: "foodblogger"}
```
Comment (shannonwells) [inline scripts/populate.mjs:291]: should show in the feed.

File: scripts/populate.mjs [hunk 6]
```diff
@@ -293,6 +316,7 @@ const accounts = [
     ],
     text:
       "Darn it, Jim, I'm a family doctor, not a swearing doctor! For pity's sake Jim, this is prime time!",
+    tag: [{type: "Mention", name: "restaurant"}]
```
Comment (shannonwells) [inline scripts/populate.mjs:319]: should NOT show up in the feed, because it's a mention and not a hashtag.

File: config/env.js [hunk 7]
```diff
@@ -87,6 +85,7 @@ function getClientEnvironment(publicUrl) {
         WDS_SOCKET_HOST: process.env.WDS_SOCKET_HOST,
         WDS_SOCKET_PATH: process.env.WDS_SOCKET_PATH,
         WDS_SOCKET_PORT: process.env.WDS_SOCKET_PORT,
+        APP_DEFAULT_TAGS: process.env.APP_DEFAULT_TAGS,
```
Comment (wilwade) [inline config/env.js:88]: @shannonwells Just a note that you don't have to have this line if you do prefix with `REACT_APP_`. So `REACT_APP_DEFAULT_TAGS` would have just worked thanks to some magic.

**Pull request merged**